### PR TITLE
ERC725: Attestations and users

### PR DESCRIPTION
**Pull request by tyleryasaka**
### Checklist:

- [x] Code contains relevant tests for the problem you are solving
- [x] Ensure all new and existing tests pass
- [x] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/docs)
- [x] Submit to the `develop` branch instead of `master`

### Description:

This adds 2 new resources, `Attestations` and `Users`. They are extensively documented in my [docs PR](https://github.com/OriginProtocol/docs/pull/22).

Some other things going on:
- ERC725 contract restructuring
- Fixes some migration issues that were making my tests flaky (not sure if anyone else has been experiencing this)
- Extensive test coverage for the attestations and users resources
- Adds a couple of conveniences to the contract service for the new web3 setup

### Example usage in demo-dapp

Can use `attestationServerUrl: "http://bridge-server-test.herokuapp.com/api/attestations"` in config. That's a heroku app that's up and running right now.

Just to make this easier to review and understand.

Can stick this in `src/services/origin.js` (in `demo-dapp`):

```
import Origin from 'origin'

let config = {}

config.ipfsDomain = process.env.IPFS_DOMAIN ? process.env.IPFS_DOMAIN : null
config.ipfsApiPort = process.env.IPFS_API_PORT ? process.env.IPFS_API_PORT : null
config.ipfsGatewayPort = process.env.IPFS_GATEWAY_PORT ? process.env.IPFS_GATEWAY_PORT : null
config.ipfsGatewayProtocol = process.env.IPFS_GATEWAY_PROTOCOL ? process.env.IPFS_GATEWAY_PROTOCOL : null
config.attestationServerUrl = "http://bridge-server-test.herokuapp.com/api/attestations"

const origin = new Origin(config)
window.web3 = origin.contractService.web3

let sendCode = () => {
  origin.attestations.phoneGenerateCode({
    phone: "555-555-5555"
  })
}

let createUser = async () => {
  let phoneAttestation = await origin.attestations.phoneVerify({
    phone: "555-555-5555",
    code: "332308"
  })
  let myNewUser = {
    profile: { claims: { name: "Wonder Woman" } },
    attestations: [ phoneAttestation ]
  }
  let user = await origin.users.set(myNewUser)
  console.log('user created', user)
}

// get your verification code, then remove me
sendCode()

// uncomment me once you have your code
// createUser()

export default origin
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,9 +33,11 @@
     "babel-polyfill": "^6.26.0",
     "bs58": "^4.0.1",
     "cross-fetch": "^2.1.1",
+    "ethereumjs-util": "^5.2.0",
```
Comment (nick) [inline package.json]: Do we still need this?

File: src/contract-service.js [hunk 2]
```diff
@@ -104,13 +110,28 @@ class ContractService {
...
+  async deploy(contract, args, options) {
+    let deployed = await this.deployed(contract)
+    let transaction = await new Promise((resolve, reject) => {
+      deployed
+        .deploy({
+          data: contract.bytecode,
+          arguments: args
+        })
+        .send(options)
+        .on("receipt", receipt => {
```
Comment (nick) [inline src/contract-service.js]: Should we listen for the "confirmation" event instead?

File: package.json (hunk 1 above)
Comment (tyleryasaka) [reply package.json]: @nick No I don't think it's needed. It's currently being used for attestation signature validation. https://github.com/OriginProtocol/origin-js/blob/d18dddb8f6c8348896d5851d8da5d6c750b79e31/src/resources/users.js#L189

We should be able to do all this with web3. Created an issue for it so I don't forget. https://github.com/OriginProtocol/origin-js/issues/153

File: src/contract-service.js (hunk 2 above)
Comment (tyleryasaka) [reply src/contract-service.js]: @nick I'm actually not that familiar with this API. I just looked at what we're doing in `ResourceBase.contractFn` with `waitTransactionFinished` and did the same thing here.

https://github.com/OriginProtocol/origin-js/blob/master/src/ResourceBase.js#L24

**Comment by wanderingstan** [comment]
@tyleryasaka one nit-pick: when running `npm test` its spitting out multiple screenfuls of hex. I wonder if some debugging code got left in? 
image

**Comment by tyleryasaka** [comment]
@wanderingstan That output is coming from the new web3 version 1.0, rather than anything I changed. The various `console.log` statements scattered through this codebase are spiting out more verbose items now.

I think we should remove `console.log` statements from this package entirely. I don't think it's good practice in general.

**Pull request merged**